src/main/java/org/sonarlint/intellij/analysis/JavaAnalysisConfigurator.java

```diff
@@ -50,118 +47,42 @@
+  static final String JAVA_TEST_BINARIES_PROPERTY = "sonar.java.test.binaries";
 
   private static final char SEPARATOR = ',';
 
   private static final String JAR_REGEXP = "(.*)!/";
   private static final Pattern JAR_PATTERN = Pattern.compile(JAR_REGEXP);
 
-  public AnalysisResults analyzeModule(Module module, Collection<VirtualFile> filesToAnalyze, IssueListener listener) {
-    Project p = module.getProject();
-    SonarLintConsole console = SonarLintConsole.get(p);
-    SonarLintServerManager core = SonarLintUtils.get(SonarLintServerManager.class);
-
-    // Configure plugin properties
-    Map<String, String> pluginProps = new HashMap<>();
-    configureModuleSettings(module, pluginProps);
-
-    // configure files
-    ModuleRootManager moduleRootManager = ModuleRootManager.getInstance(module);
-    List<ClientInputFile> inputFiles = getInputFiles(p, moduleRootManager, filesToAnalyze);
-
-    // Analyze
-    long start = System.currentTimeMillis();
-
-    SonarLintFacade facade = core.getFacadeForAnalysis(module.getProject());
-    if (facade == null) {
-      console.info("Failed to create SonarLint engine for module '" + module.getName() + "'");
-      return null;
-    }
-
-    String what;
-    if (filesToAnalyze.size() == 1) {
-      what = "'" + filesToAnalyze.iterator().next().getName() + "'";
-    } else {
-      what = Integer.toString(filesToAnalyze.size()) + " files";
-    }
-
-    console.info("Analysing " + what + "...");
-    AnalysisResults result = facade.startAnalysis(inputFiles, listener, pluginProps);
-    console.debug("Done in " + (System.currentTimeMillis() - start) + "ms\n");
-    return result;
-  }
-
-  private static Charset getEncoding(Project p, @Nullable VirtualFile f) {
-    if (f != null) {
-      Charset encoding = EncodingProjectManager.getInstance(p).getEncoding(f, true);
-      if (encoding != null) {
-        return encoding;
-      }
-    }
-    return Charset.defaultCharset();
-  }
-
-  private static List<ClientInputFile> getInputFiles(Project p, ModuleRootManager moduleRootManager, Collection<VirtualFile> filesToAnalyze) {
-    Collection<String> testFolderPrefix = findTestFolderPrefixes(moduleRootManager);
-    List<ClientInputFile> inputFiles = new LinkedList<>();
-
-    for (VirtualFile f : filesToAnalyze) {
-      boolean test = isTestFile(testFolderPrefix, f);
-      Charset charset = getEncoding(p, f);
-      inputFiles.add(new DefaultInputFile(f, test, charset));
-    }
-
-    return inputFiles;
-  }
-
-  private static boolean isTestFile(Collection<String> testFolderPrefix, VirtualFile f) {
-    String filePath = f.getPath();
-    for (String testPrefix : testFolderPrefix) {
-      if (filePath.startsWith(testPrefix)) {
-        return true;
-      }
-    }
-    return false;
-  }
-
-  private static void configureModuleSettings(@NotNull Module ijModule, @NotNull Map<String, String> properties) {
+  public Map<String, String> configure(@NotNull Module ijModule) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add the "@Override" annotation above this method signature [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1161)